homeassistant/components/linear_garage_door/cover.py

```diff
@@ -0,0 +1,164 @@
+                    )
+                )
+
+    async_add_entities(device_list)
+
+
+class LinearCoverEntity(CoordinatorEntity[LinearUpdateCoordinator], CoverEntity):
+    """Representation of a Linear cover."""
+
+    _attr_supported_features = CoverEntityFeature.OPEN | CoverEntityFeature.CLOSE
+
+    def __init__(
+        self,
+        device_id: str,
+        device_name: str,
+        subdevice: str,
+        config_entry: ConfigEntry,
+        coordinator: LinearUpdateCoordinator,
+    ) -> None:
+        """Init with device ID and name."""
+        super().__init__(coordinator)
+
+        self._attr_has_entity_name = True
+        self._device_id = device_id
+        self._device_name = device_name
+        self._subdevice = subdevice
+        self._attr_device_class = CoverDeviceClass.GARAGE
+        self._attr_unique_id = f"{device_id}-{subdevice}"
+        self._attr_name = subdevice
+        self._config_entry = config_entry
+
+    @property
+    def device_info(self) -> DeviceInfo:
+        """Return device info of a garage door."""
+        return DeviceInfo(
+            identifiers={(DOMAIN, self._device_id)},
+            name=self._device_name,
+            default_manufacturer="Linear",
+            default_model="Garage Door Opener",
+        )
+
+    @property
+    def is_closed(self) -> bool:
+        """Return if cover is closed."""
+        return bool(
+            self.coordinator.data[self._device_id]["subdevices"][self._subdevice][
+                "Open_B"
+            ]
+            == "false"
+        )
+
+    @property
+    def is_opened(self) -> bool:
+        """Return if cover is open."""
+        return bool(
+            self.coordinator.data[self._device_id]["subdevices"][self._subdevice][
+                "Open_B"
+            ]
+            == "true"
+        )
+
+    @property
+    def is_opening(self) -> bool:
+        """Return if cover is opening."""
+        return bool(
+            self.coordinator.data[self._device_id]["subdevices"][self._subdevice].get(
+                "Opening_P"
+            )
+            == "0"
+        )
+
+    @property
+    def is_closing(self) -> bool:
+        """Return if cover is closing."""
+        return bool(
+            self.coordinator.data[self._device_id]["subdevices"][self._subdevice].get(
+                "Closing_P"
+            )
+            == "100"
+        )
+
+    async def async_close_cover(self, **kwargs: Any) -> None:
+        """Close the garage door."""
+        if self.is_closed:
+            return
+
+        linear = Linear()
+
+        await linear.login(
+            email=self._config_entry.data["email"],
+            password=self._config_entry.data["password"],
+            device_id=self._config_entry.data["device_id"],
+            client_session=async_get_clientsession(self.hass),
+        )
+
+        await linear.operate_device(self._device_id, self._subdevice, "Close")
+        await linear.close()
```

Comment:
Again login each time you need data / need to perform an action seems very redundant and seems like how you would get your account banned. Every 10 seconds you are logging in to update for the coordinator and then everytime you open or close it you are logging in

Reply:
I'd create one instance of Linear() that holds email,pass, client session, and a list of device ids or whatever other information you need and I would add this to the coordinator

Then whenever you need to do anything, you could do self.coordinator.linear_api rather than repeating this code so much and logging in so many times.

Reply:
Hi Lash,

I don't believe my account would get banned, as I have been testing this with my own doors for about 2 months now and I haven't received any disciplinary action.

If I could, I would make this a Cloud Push integration, but there's a weird bug where Linear just stops sending events entirely.

I'm worried the same would happen if I had a constant WebSocket open, it would stop sending events and it would break.

Reply:
Right now you haven't. But unless if i am misreading something you are logging in 17,280 times at a minimum each day since you log in each time the coordinator updates. and the coordinator is set to update every 5 seconds

If 100 people used this integration that's almost 2 million additional log ins a day, 1000, almost 20 million. 

They will probably have to investigate why there is a massive increase in usage of their authentication server and why it is under load. Then they will either ban accounts or build some functionality to stop you from doing it how you are doing it right now.

That is my opinion at least.

Reply:
Given that they haven't updated the app in over 2 years, I think the team that develops the app doesn't care for their app anymore :(

Reply:
I agree this is a valid concern, and I also don't think there's any reason to poll the state of a garage door every 5s.
Please decrease it to something reasonable, maybe poll once per minute?